mutator.go

```diff
@@ -148,11 +151,23 @@ func (m *Mutator) ExecuteWithWriter(Writer io.Writer) error {
 			// we can't early exit, due to abstraction we have to conclude the elaboration to drain all dedupers
 			continue
 		}
-		_, err := Writer.Write([]byte(value + "\n"))
-		m.payloadCount++
+		if maxFileSize <= 0 {
+			// drain all dedupers when max-file size reached
+			continue
+		}
+		outputData := []byte(value + "\n")
+		if len(outputData) > maxFileSize {
+			// truncate output data if it exceeds maxFileSize
+			outputData = outputData[:maxFileSize]
+		}
+
```

Comment:
@RamanaReddy0M instead of truncating subdomain i think it makes more sense to ignore that value
ex: if generated domain is `api.hackerone.com`. ^ will return something like `api.` which is invalid domain and shouldn't be the case.
suggesting we ignore that value completely

```go
if len(outputData) > maxFileSize {
    maxFileSize = 0
    continue
}
```